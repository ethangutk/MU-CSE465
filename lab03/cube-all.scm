; (map (lambda (x) (* x x x x)) L)
; 
; Write a function that uses the schem 'map' higher 
;; order function in order to compute the cubed value 
;; of every element of a list. 
;;

;;  You MUST use map -- NOT recursion -- to solve this. 

(define (cube-all lst) 
  (map (lambda (x) (* x x x)) lst)
)

;; Test cases
(define (println s) (display s) (newline))

(println (cube-all '(1 2 3 4)))
(println (cube-all '()))
(println (cube-all '(0 -1 1 -2 2 3 -3)))

