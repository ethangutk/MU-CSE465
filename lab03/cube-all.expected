(1 8 27 64)
()
(0 -1 1 -8 8 27 -27)
